scripts/workers/code_upload_submission_worker.py

```diff
@@ -469,48 +469,72 @@ def update_failed_jobs_and_send_logs(
+                                )
                             )
-                        )
-                        pod_log = pod_log_response.data.decode("utf-8")
-                        submission_data = {
-                            "challenge_phase": phase_pk,
-                            "submission": submission_pk,
-                            "stdout": "",
-                            "stderr": pod_log,
-                            "submission_status": "FAILED",
-                            "result": "[]",
-                            "metadata": "",
-                        }
-                        response = evalai.update_submission_data(
-                            submission_data, challenge_pk, phase_pk
-                        )
-                        print(response)
-                    except client.rest.ApiException as e:
-                        logger.exception(
-                            "Exception while reading Job logs {}".format(e)
-                        )
+                            pod_log = pod_log_response.data.decode("utf-8")
+                            submission_data = {
+                                "challenge_phase": phase_pk,
+                                "submission": submission_pk,
+                                "stdout": "",
+                                "stderr": pod_log,
+                                "submission_status": "FAILED",
+                                "result": "[]",
+                                "metadata": "",
+                            }
+                            response = evalai.update_submission_data(
+                                submission_data, challenge_pk, phase_pk
+                            )
+                            print(response)
+                        except client.rest.ApiException as e:
+                            logger.exception(
+                                "Exception while reading Job logs {}".format(e)
+                            )
+    except Exception as e:
+        logger.exception("Exception while reading Job {}".format(e))
+        submission_data = {
+            "challenge_phase": phase_pk,
+            "submission": submission_pk,
+            "stdout": "",
+            "stderr": "Submission Job failed.",
+            "submission_status": "failed",
+            "result": "[]",
+            "metadata": "",
+        }
+        response = evalai.update_submission_data(
+            submission_data, challenge_pk, phase_pk
+        )
+        message_receipt_handle = message.get("receipt_handle")
+        try:
+            delete_job(api_instance, job_name)
+            evalai.delete_message_from_sqs_queue(message_receipt_handle)
+        except Exception as e:
+            logger.exception("Failed to delete submission job: {}".format(e))
+            # Delete message from sqs queue to avoid re-triggering job delete
+            evalai.delete_message_from_sqs_queue(message_receipt_handle)
```

Comment:
@savish28 can we move this logic of marking submission failed, deleting job and deleting message to a separate method named `cleanup_submission` where you pass the required details. The current code is not readable.

Reply:
Yeah, makes sense.

Reply:
@Ram81 Updated the code, Do we want something like this.